src/objects/value-serializer.cc

```diff
@@ -426,6 +426,8 @@ std::pair<uint8_t*, size_t> ValueSerializer::Release() {
   buffer_ = nullptr;
   buffer_size_ = 0;
   buffer_capacity_ = 0;
+  REPLAY_ASSERT_MAYBE_EVENTS_DISALLOWED(
+    "[TT-1403] V8ScriptValueSerializer::Serialize %zu", buffer_size_);
```

Comment:
line 427 sets `buffer_size_` to 0, so you probably want to move this up to the top of the function or change `buffer_size_` here to `result.second`